runtime/src/utxo.rs

```diff
@@ -1,20 +1,30 @@
-use super::Aura;
 use codec::{Decode, Encode};
 use frame_support::{
 	decl_event, decl_module, decl_storage,
 	dispatch::{DispatchResult, Vec},
 	ensure,
 };
-use sp_core::{H256, H512};
+//TODO is it correct to use the Public trait here??
+// ultimately I'm just trying to call as_slice to convert a pubkey to a H256
+use sp_core::{H256, H512, crypto::Public as _};
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
 use sp_core::sr25519::{Public, Signature};
```

Comment:
Ah its already imported on line 12. Can we clean up the imports a bit and merge the things we're importing from the same crates? Also, items from `sp_runtime` is being imported in two separate lines as well. Lets combine?

Reply:
line 12 imports this [`Public` struct](https://substrate.dev/rustdocs/master/sp_application_crypto/sr25519/struct.Public.html), but it seems I also need this [`Public` trait](https://substrate.dev/rustdocs/master/sp_core/crypto/trait.Public.html) in order to convert from pubkey to H256.

Without it I get:
```
error[E0599]: no method named `as_slice` found for reference `&sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::sr25519::Public` in the current scope
   --> /home/joshy/ProgrammingProjects/utxo-workshop/runtime/src/utxo.rs:245:36
    |
245 |             pubkey: H256::from_slice(author.as_slice()),
    |                                             ^^^^^^^^ method not found in `&sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::sr25519::Public`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use crate::sp_api_hidden_includes_construct_runtime::hidden_include::sp_runtime::sp_application_crypto::sp_core::crypto::Public;`
```

This is also why I wondered whether I was making that conversion in the correct way.

I totally agree about the import style in general. Cleaning it up now 🧹